src/chromatix/utils/czt.py

```diff
@@ -0,0 +1,193 @@
+from typing import List
+
+import jax.numpy as jnp
+from chex import Array
+
+
+def _verify_axes(x, axes):
+    axes = [a + x.ndim if a < 0 else a for a in axes]
+    if any(a >= x.ndim or a < 0 for a in axes):
+        raise ValueError("axes exceeds dimensionality of input")
+    if len(set(axes)) != len(axes):
+        raise ValueError("all axes must be unique")
+    return axes
+
+
+def _verify_cztn_input(x, a, w, m, axes):
+    D = len(a)
+    if not (len(w) == len(m) == D):
+        raise ValueError("Length of [a], [w], and [m] must match.")
+
+    if x.ndim < D:
+        raise ValueError("[x] does not have enough dimensions.")
+
+    if axes is not None:
+        assert len(axes) == D, "Length of [axes] must match [a], [w], and [m]."
+        axes = _verify_axes(x, axes)
+    else:
+        axes = list(range(D))
+
+    # check valid values
+    for d in range(D):
+        a[d] = complex(a[d])
+        w[d] = complex(w[d])
+        if not jnp.isclose(abs(a[d]), 1):
+            raise ValueError(
+                "Parameter[a[d]] must lie on the unit circle for numerical stability."
+            )
+        if not jnp.isclose(abs(w[d]), 1):
+            raise ValueError("Parameter[w[d]] must lie on the unit circle.")
+        if m[d] <= 0:
+            raise ValueError("Parameter[m[d]] must be positive.")
+
+    return axes, a, w
+
+
+def czt(x: Array, m: int, a: complex, w: complex, axis=-1) -> Array:
+    """
+    Chirp Z-transform (CZT) of a signal along one dimension. The CZT is a
+    generalization of the discrete Fourier transform (DFT). The DFT samples the
+    Z plane at uniformly-spaced points on the unit circle, whereas the CZT
+    samples the Z plane at uniformly-spaced points on a spiral. This can be
+    used to interpolate the DFT to any desired frequency resolution.
+
+    Bluestein's algorithm is used to compute the CZT as a convolution.
+
+    !!! warning
+        Using float32/complex64 may have numerical accuracy issues.
+
+    Args:
+        x: Input signal to transform.
+        m: Number of samples in the output.
+        a: The starting point in the complex plane.
+        w: The ratio between points in each step.
+    """
+    axes, a, w = _verify_cztn_input(x, a=[a], w=[w], m=[m], axes=[axis])
+    axis = axes[0]
+    a = a[0]
+    w = w[0]
```

Comment:
First you turn it into a list, than unpack again. Is that necessary?

Reply:
It isn't, I've removed and added some `checkify` checks